cli/cmd/env/pull.go

```diff
@@ -76,8 +76,10 @@ func PullCmd(ch *cmdutil.Helper) *cobra.Command {
 				resVars[v.Name] = v.Value
 			}
 
+			dotEnv := parser.GetDotEnv()
+
 			// If the variables match any existing .env file, do nothing
-			if maps.Equal(resVars, parser.DotEnv) {
+			if reflect.DeepEqual(resVars, dotEnv) {
```

Comment:
What is wrong with `maps.Equal`? The members are `string`s, so don't require deep equality checks